fdupes.c

```diff
@@ -295,60 +298,67 @@ int grokdir(char *dir, file_t **filelistp)
       newfile->d_name = (char*)malloc(strlen(dir)+strlen(dirinfo->d_name)+2);
 
       if (!newfile->d_name) {
-	errormsg("out of memory!\n");
-	free(newfile);
-	closedir(cd);
-	exit(1);
+        errormsg("out of memory!\n");
+        free(newfile);
+        closedir(cd);
+        exit(1);
       }
 
       strcpy(newfile->d_name, dir);
       lastchar = strlen(dir) - 1;
       if (lastchar >= 0 && dir[lastchar] != '/')
-	strcat(newfile->d_name, "/");
+        strcat(newfile->d_name, "/");
       strcat(newfile->d_name, dirinfo->d_name);
-      
+
       if (ISFLAG(flags, F_EXCLUDEHIDDEN)) {
-	fullname = strdup(newfile->d_name);
-	name = basename(fullname);
-	if (name[0] == '.' && strcmp(name, ".") && strcmp(name, "..") ) {
-	  free(newfile->d_name);
-	  free(newfile);
-	  continue;
-	}
-	free(fullname);
+        fullname = strdup(newfile->d_name);
+        name = basename(fullname);
+        if (name[0] == '.' && strcmp(name, ".") && strcmp(name, "..") ) {
+          free(newfile->d_name);
+          free(newfile);
+          continue;
+        }
+        free(fullname);
       }
 
       if (filesize(newfile->d_name) == 0 && ISFLAG(flags, F_EXCLUDEEMPTY)) {
-	free(newfile->d_name);
-	free(newfile);
-	continue;
+        free(newfile->d_name);
+        free(newfile);
+        continue;
       }
 
       if (stat(newfile->d_name, &info) == -1) {
-	free(newfile->d_name);
-	free(newfile);
-	continue;
+        free(newfile->d_name);
+        free(newfile);
+        continue;
       }
 
       if (lstat(newfile->d_name, &linfo) == -1) {
-	free(newfile->d_name);
-	free(newfile);
-	continue;
+        free(newfile->d_name);
+        free(newfile);
+        continue;
+      }
+
+      if (S_ISREG(linfo.st_mode) &&
+          ((filesize(newfile->d_name) < min_size ) || (filesize(newfile->d_name) > max_size ))) {
```

Comment:
You should only call `filesize` once because each call results in one `stat` call.

Reply:
good point! fixed.

Reply:
Maybe @adrianlopezroche wants to review further if this is still relevant.